perf/benchmark/security/generate_policies/generate_policies.go

```diff
@@ -185,14 +210,111 @@ func generatePeerAuthentication(policyData SecurityPolicy, policyHeader *MyPolic
 	return yaml, nil
 }
 
+func generateToken(policyData SecurityPolicy, privateKey *rsa.PrivateKey) (string, error) {
+	issuer := fmt.Sprintf("issuer-%d", policyData.RequestAuthN.NumJwks)
+	if policyData.RequestAuthN.TokenIssuer != "" {
+		issuer = policyData.RequestAuthN.TokenIssuer
+	}
+	token := jwt.NewWithClaims(jwt.SigningMethodRS256, jwt.MapClaims{
+		"iss": issuer,
+		"sub": "subject",
+	})
+	if policyData.RequestAuthN.InvalidToken {
+		newPrivateKey, err := rsa.GenerateKey(rand.Reader, 2048)
+		if err != nil {
+			return "", err
+		}
+		privateKey = newPrivateKey
+	}
+	tokenString, err := token.SignedString(privateKey)
+	if err != nil {
+		return "", err
+	}
+	return tokenString, nil
+}
+
+func generateJwksBytes(privateKey *rsa.PrivateKey) ([]byte, error) {
+	jwks := &Jwks{
+		Keys: []*Jwk{
+			{
+				E:   base64.URLEncoding.EncodeToString(big.NewInt(int64(privateKey.PublicKey.E)).Bytes()),
+				N:   base64.URLEncoding.EncodeToString((*privateKey.PublicKey.N).Bytes()),
+				Kty: "RSA",
+			},
+		},
+	}
+
+	jwksBytes, err := json.Marshal(jwks)
+	if err != nil {
+		return nil, err
+	}
+	return jwksBytes, nil
+}
+
+func writeTokenIntoFile(token string, fileName string) error {
+	file, err := os.Create(fileName)
+	if err != nil {
+		return err
+	}
+	_, err = file.WriteString(fmt.Sprintf(`"Authorization":"Bearer %s"`, token))
+	if err != nil {
+		file.Close()
+		return err
+	}
+	err = file.Close()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func generateRequestAuthentication(policyData SecurityPolicy, policyHeader *MyPolicy) (string, error) {
+	privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return "", err
+	}
+	token, err := generateToken(policyData, privateKey)
+	if err != nil {
+		return "", err
+	}
+	err = writeTokenIntoFile(token, "token.txt")
+	if err != nil {
+		return "", err
+	}
+	jwksBytes, err := generateJwksBytes(privateKey)
```

Comment:
you can change generateJwksBytes to return string directly, and then rename it to just generateJwks

Reply:
Great done